instrumentation/vaadin-14.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vaadin/VaadinTracer.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.vaadin;
+
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.router.Location;
+import com.vaadin.flow.server.RequestHandler;
+import com.vaadin.flow.server.VaadinService;
+import com.vaadin.flow.server.communication.rpc.RpcInvocationHandler;
+import elemental.json.JsonObject;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.instrumentation.api.servlet.ServletContextPath;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import io.opentelemetry.instrumentation.api.tracer.ServerSpan;
+import java.lang.reflect.Method;
+
+public class VaadinTracer extends BaseTracer {
+  private static final ContextKey<VaadinServiceContext> SERVICE_CONTEXT_KEY =
+      ContextKey.named("opentelemetry-vaadin-service-context-key");
+  private static final ContextKey<Object> REQUEST_HANDLER_CONTEXT_KEY =
+      ContextKey.named("opentelemetry-vaadin-request-handler-context-key");
+
+  private static final VaadinTracer TRACER = new VaadinTracer();
+
+  public static VaadinTracer tracer() {
+    return TRACER;
+  }
+
+  private VaadinTracer() {
+    super(GlobalOpenTelemetry.get());
+  }
```

Comment:
@anuraaga What is our current preferred form of tracer constructors?

Reply:
Ah sorry for late reply. I don't think it matters too much for Javaagent instrumentation so this seems ok.